src/lib.rs

```diff
@@ -37,11 +37,26 @@ mod errors {
                 description("unrecognizable charset")
                 display("uchardet was unable to recognize a charset")
             }
-            DataHandlingError {
-                description("data handling error")
-                display("uchardet failed to handle the supplied data (out of memory?)")
+            OutOfMemory {
+                description("out of memory error")
+                display("uchardet ran out of memory")
+            }
+            Other(int: i32) {
+                description("unknown error")
+                display("uchardet returned unknown error {}", int)
+            }
+        }
+    }
+
+    impl ErrorKind {
+        pub fn from_nsresult(nsresult: ::ffi::nsresult) -> ErrorKind {
```

Comment:
I don't know how to make this private but still accessible for `EncodingDetector::handle_data`.